graph-partitioning/graph_partitioning.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2019 D-Wave Systems, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ------ Import necessary packages ----
+import networkx as nx 
+from collections import defaultdict
+from itertools import combinations
+from dwave.system.samplers import DWaveSampler
+from dwave.system.composites import EmbeddingComposite
+
+# ------- Set tunable parameters -------
+num_reads = 1000
+gamma = 156
+
+# ------- Set up our graph -------
+G = nx.gnp_random_graph(40, 0.2)
+
+print("\nGraph on",len(G.nodes),"nodes created with",len(G.edges),"out of",len(G.nodes)*(len(G.nodes)-1)/2,"possible edges.\n")
+
+# ------- Set up our QUBO dictionary -------
+
+# Initialize our Q matrix
+Q = defaultdict(int)
+
+# Fill in Q matrix
+for u, v in G.edges:
+    Q[(u,u)] += 1
+    Q[(v,v)] += 1
+    Q[(u,v)] += -2
+
+for i in G.nodes:
+    Q[(i,i)] += gamma*(1-1*len(G.nodes))
+
+for i,j in combinations(G.nodes,2):
+	Q[(i,j)] += 2*gamma
+
+# ------- Run our QUBO on the QPU -------
+
+# Set chain strength
+chain_strength = gamma*len(G.nodes)
+
+# Run the QUBO on the solver from your config file
+sampler = EmbeddingComposite(DWaveSampler())
+response = sampler.sample_qubo(Q, chain_strength=chain_strength, num_reads=num_reads)
+
+# See if the best solution found is feasible, and if so print the number of cut edges.
+sample = response.record.sample[0]
+if sum(sample) == 20:
+    num_cut_edges = 0
+    for u,v in G.edges:
+        num_cut_edges += sample[u]+sample[v]-2*sample[u]*sample[v]
+    print("Valid partition found with",num_cut_edges,"cut edges.\n")
```

Comment:
don't need `"\n"`. probably want to add a whitespace after the commas.

Reply:
Fixed